Binary files ACS_hostapd-2.6_new3/hostapd/hostapd_debug and ACS_hostapd-2.6_new2/hostapd/hostapd_debug differ
diff --exclude 'msvc*' -Npur ACS_hostapd-2.6_new3/src/ap/beacon.c ACS_hostapd-2.6_new2/src/ap/beacon.c
--- ACS_hostapd-2.6_new3/src/ap/beacon.c	2017-01-18 18:58:17.000000000 +0200
+++ ACS_hostapd-2.6_new2/src/ap/beacon.c	2017-01-18 18:51:22.000000000 +0200
@@ -1203,14 +1203,8 @@ int ieee802_11_build_ap_params(struct ho
 	params->basic_rates = hapd->iface->basic_rates;
 	params->ssid = hapd->conf->ssid.ssid;
 	params->ssid_len = hapd->conf->ssid.ssid_len;
-	if ((hapd->conf->wpa & (WPA_PROTO_WPA | WPA_PROTO_RSN)) ==
-	    (WPA_PROTO_WPA | WPA_PROTO_RSN))
-		params->pairwise_ciphers = hapd->conf->wpa_pairwise |
-			hapd->conf->rsn_pairwise;
-	else if (hapd->conf->wpa & WPA_PROTO_RSN)
-		params->pairwise_ciphers = hapd->conf->rsn_pairwise;
-	else if (hapd->conf->wpa & WPA_PROTO_WPA)
-		params->pairwise_ciphers = hapd->conf->wpa_pairwise;
+	params->pairwise_ciphers = hapd->conf->wpa_pairwise |
+				   hapd->conf->rsn_pairwise;
 	params->group_cipher = hapd->conf->wpa_group;
 	params->key_mgmt_suites = hapd->conf->wpa_key_mgmt;
 	params->auth_algs = hapd->conf->auth_algs;
diff --exclude 'msvc*' -Npur ACS_hostapd-2.6_new3/src/ap/ieee802_11.c ACS_hostapd-2.6_new2/src/ap/ieee802_11.c
--- ACS_hostapd-2.6_new3/src/ap/ieee802_11.c	2017-01-18 18:59:06.000000000 +0200
+++ ACS_hostapd-2.6_new2/src/ap/ieee802_11.c	2017-01-18 18:56:32.000000000 +0200
@@ -2223,6 +2223,18 @@ static void handle_assoc(struct hostapd_
 		return;
 	}
 
+	if ((!hapd->conf->ieee802_1x && !hapd->conf->wpa && !hapd->conf->osen) ||
+	    (sta && (sta->auth_alg == WLAN_AUTH_FT))) {
+	    /*
+	    * Open, static WEP, or FT protocol; no separate authorization
+	    * step.
+	    */
+	    ap_sta_set_authorized(hapd, sta, 1);
+	}
+	else {
+	    ap_sta_set_authorized(hapd, sta, 0);
+	}
+
 	if ((fc & WLAN_FC_RETRY) &&
 	    sta->last_seq_ctrl != WLAN_INVALID_MGMT_SEQ &&
 	    sta->last_seq_ctrl == seq_ctrl &&
